### ISSUE TITLE
boofuzz crashes in python2

### ISSUE THREAD

**Issue by secenv** [jtpereyda/boofuzz#351]
Script and logs shown below. I'm not sure if it's related to the 's_from_file' primitive. It works well when boofuzz is installed through pip3.

```
$ echo -e 'a\nb' > test.txt
$ cat test.py
from boofuzz import *

session = Session( target=Target( connection=SocketConnection( host="127.0.0.1", port=12345, proto='tcp', server=True ),),)
s_initialize("A")
s_from_file("0", filename="test.txt") # same happens with 'encoding="utf8"'
session.connect(session.root, s_get("A"))
session.fuzz()

$ python test.py
[2019-11-20 14:48:57,901]     Info: Web interface can be found at http://localhost:26000
[2019-11-20 14:48:57,903] Test Case: 1: A.no-name.1
[2019-11-20 14:48:57,903]     Info: Type: FromFile. Default value: '0'. Case 1 of 2 overall.
[2019-11-20 14:48:57,903]     Info: Opening target connection (127.0.0.1:12345)...
[2019-11-20 14:48:59,064]     Info: Connection opened.
[2019-11-20 14:48:59,064]   Test Step: Fuzzing Node 'A'
[2019-11-20 14:48:59,064]     Info: Sending 2 bytes...
[2019-11-20 14:48:59,085]     Error!!!! Unexpected exception! Traceback (most recent call last):
                                File "/home/user/.local/lib/python2.7/site-packages/boofuzz/sessions.py", line 703, in _main_fuzz_loop
                                  self._fuzz_current_case(*fuzz_args)
                                File "/home/user/.local/lib/python2.7/site-packages/boofuzz/sessions.py", line 1431, in _fuzz_current_case
                                  self.transmit_fuzz(target, self.fuzz_node, path[-1], callback_data=callback_data)
                                File "/home/user/.local/lib/python2.7/site-packages/boofuzz/sessions.py", line 1114, in transmit_fuzz
                                  self.targets[0].send(data)
                                File "/home/user/.local/lib/python2.7/site-packages/boofuzz/sessions.py", line 157, in send
                                  self._fuzz_data_logger.log_send(data[:num_sent])
                                File "/home/user/.local/lib/python2.7/site-packages/boofuzz/fuzz_logger.py", line 71, in log_send
                                  fuzz_logger.log_send(data=data)
                                File "/home/user/.local/lib/python2.7/site-packages/boofuzz/fuzz_logger_text.py", line 52, in log_send
                                  msg_type='send')
                                File "/home/user/.local/lib/python2.7/site-packages/boofuzz/fuzz_logger_text.py", line 77, in _print_log_msg
                                  print(helpers.format_log_msg(msg_type=msg_type, description=msg, data=data, indent_size=self.INDENT_SIZE),
                                File "/home/user/.local/lib/python2.7/site-packages/boofuzz/helpers.py", line 402, in format_log_msg
                                  msg = hex_to_hexstr(input_bytes=data)
                                File "/home/user/.local/lib/python2.7/site-packages/boofuzz/helpers.py", line 434, in hex_to_hexstr
                                  return hex_str(input_bytes) + " " + repr(input_bytes)
                                File "/home/user/.local/lib/python2.7/site-packages/boofuzz/helpers.py", line 348, in hex_str
                                  return ' '.join("{:02x}".format(b) for b in bytearray(s))
                              TypeError: unicode argument without an encoding
                              
Traceback (most recent call last):
  File "test.py", line 7, in <module>
    session.fuzz()
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/sessions.py", line 598, in fuzz
    self._main_fuzz_loop(self._iterate_protocol())
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/sessions.py", line 703, in _main_fuzz_loop
    self._fuzz_current_case(*fuzz_args)
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/sessions.py", line 1431, in _fuzz_current_case
    self.transmit_fuzz(target, self.fuzz_node, path[-1], callback_data=callback_data)
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/sessions.py", line 1114, in transmit_fuzz
    self.targets[0].send(data)
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/sessions.py", line 157, in send
    self._fuzz_data_logger.log_send(data[:num_sent])
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/fuzz_logger.py", line 71, in log_send
    fuzz_logger.log_send(data=data)
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/fuzz_logger_text.py", line 52, in log_send
    msg_type='send')
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/fuzz_logger_text.py", line 77, in _print_log_msg
    print(helpers.format_log_msg(msg_type=msg_type, description=msg, data=data, indent_size=self.INDENT_SIZE),
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/helpers.py", line 402, in format_log_msg
    msg = hex_to_hexstr(input_bytes=data)
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/helpers.py", line 434, in hex_to_hexstr
    return hex_str(input_bytes) + " " + repr(input_bytes)
  File "/home/user/.local/lib/python2.7/site-packages/boofuzz/helpers.py", line 348, in hex_str
    return ' '.join("{:02x}".format(b) for b in bytearray(s))
TypeError: unicode argument without an encoding
```

**Comment by SR4ven** [comment]
Thanks for reporting this issue @secenv!

Have you tried saving your file with `ascii` encoding?
Also check the [docs](https://boofuzz.readthedocs.io/en/latest/user/static-protocol-definition.html#boofuzz.s_from_file) and [this](https://stackoverflow.com/questions/41542613/utf-8-string-in-python-2-and-3)

I'll see if we can fix this inside the `hex_str` function.

**Comment by secenv** [comment]
Both files used as test files are ASCII encoded:
```
$ file test.*
test.py:  Python script, ASCII text executable
test.txt: ASCII text
```

**Comment by SR4ven** [comment]
@secenv I was able to reproduce the issue and opened #352 to fix this.

Please test this solution under different circumstances (especially file encodings) and report back.

**Comment by SR4ven** [comment]
Fixed by #352
